client/src/app/components/stock-stats/stock-stats.component.ts

```diff
@@ -0,0 +1,44 @@
+import { Component, Input } from '@angular/core';
+import { Stock } from './../../interfaces/stock';
+
+@Component({
+  selector: 'app-stock-stats',
+  templateUrl: './stock-stats.component.html',
+  styleUrls: ['./stock-stats.component.scss'],
+})
+export class StockStatsComponent {
+  @Input() stockInfo: Stock;
+  constructor() {}
+
+  get open(): string  {
+    return this.stockInfo ? this.stockInfo.stats.open ? this.stockInfo.stats.open.toFixed(2) : '' : '';
```

Comment:
Using a nested ternary statement '?' is pretty tough to read and can make debugging tougher. In this case we should use if statements. Applies to the rest